### PR TITLE
Correct timestep for img2img initial noise addition

### PR DESCRIPTION
**Pull request by wfng92**
There seems to be difference between huggingface's diffusers and compvis's implementation for img2img as the [wrong timestep is used for the additive noise for image-2-image](https://github.com/CompVis/stable-diffusion/pull/533).

Summary from the relevant PR raised by the HuggingFace team:
> `t+1` is used as the timestep to add noise to the original image, but `[0, ..., t]` is used afterwards for the denoising process. We should however also use `t` when adding the noise to the original image.
> This can be quite easily verified by doing the following. Run a img2img with a small number of update steps and a very low strength because then differences between t and t+1 become quite clear.

From

```python
torch.tensor([t_enc]).to(self.model.device)
```

to

```
torch.tensor([t_enc - 1]).to(self.model.device)
```

For example, using the following `init_img` and configuration:

sketch-mountains-input

* `prompt`: A fantasy landscape, trending on artstation
* `strength`: 0.1
* `seed`: 42
* `sampler_name`: DDIM
* `steps`: 10

The existing code will generate an image with the following noise:

old_code

The new implementation will produce an image with lesser noise:

new_code

This PR affects only the img2img script. I realized that [text2img2img](https://github.com/invoke-ai/InvokeAI/blob/bd0c0d77d236ce118db1088bb3a66dee0efd83ea/ldm/invoke/generator/txt2img2img.py#L86) and [inpainting](https://github.com/invoke-ai/InvokeAI/blob/bde456f9fa2579910717566eed031e55da5b450d/ldm/invoke/generator/inpaint.py#L282) share the same code base but I am not sure if the same implementation should be applied as well.

### CODE REVIEW

**Comment by lstein** [comment]
Thanks so much for finding this problem. I will test it out the proposed solution later today.

**Review by lstein — APPROVED** [review]
Thank you for finding and fixing this. I have applied your fix to the inpaint and txt2img2img paths as well.
